### DISCUSSION TITLE
Standardize the model query builder params

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#1334] [Ideas]
In some query builder methods, you can pass camelCase params and in some you can't.

For example: you can pass camelCase inside .where, but in update you can't, you have to pass snake_keys params.
And in relationship methods you can pass camelCase params and everything is ok
```
    await account.related('profile')
      .query()
      .where({accountId: account.id}) // camelCase, ok
      .update(AccountUtils.getUpdateProfileParams(params)) // camelCase params , will throw an error
```

```
    await account.related('profile')
      .updateOrCreate({accountId: account.id},AccountUtils.getUpdateProfileParams(params)) //camelCase params, result is ok
```

**Comment by thetutlage** [comment]
Yes sounds reasonable. Mind sharing a PR for same?

**Reply by popovicieduard** [reply]
I’ll look into it and see what I can do. Just one question so it’s more clear, only the query builder uses snake_case ? Or what is the standard right now? If you could give more insight I would really appreciate it.

**Reply by thetutlage** [reply]
If you use the `Database` module directly, then you have to define all the keys as they are in your database. We perform no transformations.

If you are using Models and the Model query builder `Model.query()`, then we follow transform keys to `snake_case`. This is just a convention, since we think mostly database schemas use `snake_case` and developers writing javascript will prefer `camelCase`
